.github/workflows/tests.yml

```
@@ -5,12 +5,56 @@ on:
   # but only for the main branch
   push:
     branches:
-      - main  # Trigger on push events to the main branch
+      - main # Trigger on push events to the main branch
       - beta # Trigger on push events to the beta branch
+    # go upvote https://github.com/actions/runner/issues/1182 and yell at microsoft until they fix this or ditch yml for workflows
+    paths:
+      # src and test files
+      - "src/**"
+      - "test/**"
+      - "public/**" # PLEASE REVIEW -- Does it make sense to include this? Can modifying these break tests? (REMOVE COMMENT AFTER REVIEW)
```

```suggestion
      - "public/**"
```
There are tests to make sure there's no malformed sprite-shader json files so this does need to be included (see: <https://github.com/pagefaultgames/pokerogue/blob/beta/test/sprites/pokemonSprite.test.ts>).